src/fork_gc.c

```diff
@@ -1006,11 +1021,18 @@ static FGCError FGC_parentHandleTags(ForkGC *gc, RedisModuleCtx *rctx) {
       break;
     }
 
+    // No more tags values in tag field
     if (value == NULL) {
       RS_LOG_ASSERT(status == FGC_COLLECTED, "GC status is COLLECTED");
       break;
     }
 
+    char *tagVal;
+    size_t tagValLen;
+    if (FGC_recvBuffer(gc, (void **)&tagVal, &tagValLen) != REDISMODULE_OK) {
+      return FGC_PARENT_ERROR;
```

Comment:
Why is this a parrent error? also if you return here you do not free `header.field` no?

Reply:
Also, it's better to follow the other variables and put declaration on loop start and initialize them.